### PR TITLE
plugins/obsidian: switch to mkNeovimPlugin

### PR DESCRIPTION
**Pull request by GaetanLepage**

### CODE REVIEW

File: plugins/utils/obsidian/options.nix [hunk 1]
```diff
@@ -0,0 +1,537 @@
...
+          @return string
+        ```
+      '';
+
+    confirm_img_paste = helpers.defaultNullOpts.mkBool true ''
+      Whether to prompt for confirmation when pasting an image.
+    '';
+  };
+
+  callbacks = {
```
Comment (traxys) [inline plugins/utils/obsidian/options.nix:500]: this is missing `leave_note`

**Review by traxys — CHANGES_REQUESTED** [review]

File: plugins/utils/obsidian/default.nix [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+      ["dailyNotes" "aliasFormat"]
+      ["dailyNotes" "template"]
+      "useAdvancedUri"
+      "openAppForeground"
+      "sortBy"
+      "sortReversed"
+      "openNotesIn"
+      ["ui" "enable"]
+      ["ui" "updateDebounce"]
+      ["ui" "checkboxes"]
```
[the remark this answers is not in the document]
Comment (GaetanLepage) [reply plugins/utils/obsidian/default.nix:46]: Indeed, I marked it as "removed".

File: plugins/utils/obsidian/options.nix (hunk 1 above)
Comment (GaetanLepage) [reply plugins/utils/obsidian/options.nix:500]: Indeed, I have forgotten it. Now fixed.

File: plugins/utils/obsidian/options.nix [hunk 3]
```diff
@@ -0,0 +1,537 @@
...
+    ''
+      Configure key mappings.
+    '';
+
+  picker = {
+    name = helpers.mkNullOrOption (types.enum ["telescope.nvim" "fzf-lua" "mini.pick"]) ''
+      Set your preferred picker.
+    '';
+
+    mappings =
```
[the remark this answers is not in the document]
Comment (GaetanLepage) [reply plugins/utils/obsidian/options.nix]: Yes, the README is out of date and not consistent with the source code...

**Review by traxys — APPROVED** [review]

**Pull request merged**